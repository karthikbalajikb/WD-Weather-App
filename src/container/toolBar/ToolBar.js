import React from 'react';
import PropTypes from 'prop-types';
import WDAppBar from '../../components/appBar/AppBar';

class ToolBar extends React.Component {
  callBack = city => this.props.selectedCity(city);

  render() {
    return <WDAppBar selectedCity={this.callBack} />;
  }
}

ToolBar.propTypes = {
  selectedCity: PropTypes.func.isRequired,
};

export default ToolBar;
