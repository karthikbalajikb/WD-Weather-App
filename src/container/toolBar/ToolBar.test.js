import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ToolBar from './ToolBar';

describe('Container : Toolbar', () => {
  let mock;

  beforeEach(() => {
    mock = jest.fn();
  });

  it('ToolBar snapshot test', () => {
    const component = renderer.create(<ToolBar selectedCity={mock} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Get selected city props from WDAppBar callabck', () => {
    const wrapper = shallow(<ToolBar selectedCity={mock} />);
    wrapper.instance().callBack('Bangalore');
    expect(mock).toHaveBeenCalledWith('Bangalore');
  });
});
