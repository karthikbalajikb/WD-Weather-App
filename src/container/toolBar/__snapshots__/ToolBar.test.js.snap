// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container : Toolbar ToolBar snapshot test 1`] = `
<div
  className="WDAppBar-root-1"
>
  <header
    className="MuiPaper-root-12 MuiPaper-shadow4-18 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-10"
  >
    <div
      className="MuiToolbar-root-39 MuiToolbar-gutters-40"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42 WDAppBar-menuButton-3"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label-46"
        >
          <i
            className="material-icons"
          >
            cloud
          </i>
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
      <h2
        className="MuiTypography-root-56 MuiTypography-title-62 MuiTypography-colorInherit-75 WDAppBar-flex-2"
      >
        WD Weather App
      </h2>
      <div
        className="WDAutoSearchSelect-root-80"
      >
        <div
          className="MuiInput-root-83 MuiInput-inkbar-85 MuiInput-underline-89 WDAutoSearchSelect-width-81"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="Select MuiInput-input-92 is-clearable is-searchable Select--single"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-react-select-single--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select City
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-react-select-single--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id="react-select-single"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-98"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
